feat(portal): updated portal loader and added demo dataset

davidruvolo51 opened the pull request:
### What are the main changes you did

This PR (part of molgenis/GCC#1056) expands the existing portal loader to include settings and the option to load a demo dataset (the RD Synthetic dataset). Changes include:

- [x] feat: added option to load demo data; import settings and other files
- [x] feat: add demo data for the portal
- [x] feat: added settings to configure the menu and default user (added anonymous role)
- [x] feat: added new resources; updated organisations resource

### How to test

- Navigate to the preview and sign in as admin
- Create a new schema using the `RD3_V2` template and include the demo data
- View the tables to see the demo data

### Checklist
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

mswertz reviewed (changes requested):
Great, just a few changes in the area of the catalogue representation are desired. 

It would be cool to also populate a catalogue that contains these as resources.

I see you have a different understanding of the catalogue model as compared to how the catalogue team is using it. I highlighted the main issues but it is probably best to talk about this. Meanwhile if you allow me I would be interested to see if I can fix it.

davidruvolo51 commented on data/_demodata/applications/portal/Gender at birth.csv:
```diff
@@ -0,0 +1,14 @@
+name,definition,codesystem,code,ontologyTermURI
```
That's correct. It's an ontology, but in the portal, we've changed the name and the existing file conflicts with the new table name. At some point, we will introduce a breaking change to correct the gender at birth table and others.

mswertz replied:
okay, than this should be on the datamodel team agenda pls.

mswertz commented on data/_demodata/applications/portal/Datasets.csv:
```diff
@@ -0,0 +1,2 @@
+resource,name,label,date created,dataset type,description,
```
I noticed that @YpeZ used collection events for this, but I suppose datasets makes more sense
(there is a semi religious discussion wether collection events are a variant of dataset.

YpeZ replied:
True, I used that for the practical reason that _Datasets_ is not used in the UMCG catalogue, whereas _Collection events_ plays an important role there. _Datasets_ makes more sense in this instance

davidruvolo51 replied:
Populated table with EGAD metadata

willemijn-oudijk commented on data/_demodata/applications/portal/Sample preparations.csv:
Minor thing: we did include the 'included in datasets' columns to Sequencing runs, but not in here even though they both extend Protocol activity.

willemijn-oudijk commented on data/_demodata/applications/portal/Pedigree members.csv:
At lines 15 and 16 individual should be switched: it now says that Case5M is the Biological Father and Case5F is the Biological Mother, should be the other way around.

willemijn-oudijk reviewed (approved):
Looks good. I noticed two minor things, see comments.

BrendaHijmans commented on data/_demodata/applications/portal/Resources.csv:
```diff
@@ -0,0 +1,2 @@
+id,name,website,type,description
+EGA,European Genome-Phenome Archive,https://ega-archive.org,Data source,"The European Genome-phenome Archive (EGA) is a service for permanent archiving and sharing of personally identifiable genetic, phenotypic, and clinical data generated for the purposes of biomedical research projects or in the context of research-focused healthcare systems."
```
would type='Registry' fit better here?

davidruvolo51 replied:
Changed to EGAS metadata

BrendaHijmans commented on data/_ontologies/Organisations.csv:
```diff
@@ -1,27 +1,27 @@
-RD3,VASCERN,VASCERN,https://ror.org/005pygq66
-RD3,UMCG,UMCG,https://ror.org/03cv38k47
\ No newline at end of file
+ROR,ERN Board of Member States,ERN Board of Member States,https://ror.org/00r7apq26
```
why is this in ontologies? It does not fit the ontology table structure.

davidruvolo51 replied:
>why is this in ontologies? It does not fit the ontology table structure.

I think this is left over from the RD3-FDH integration. We will delete it or merge with another file in a future PR. I will add this to the ticket.

mswertz replied:
I need a little more explanation how this relates to this PR

mswertz replied:
BTW I like the ROR don't get me wrong, but why don't you simply make it follow the ontoloy structure?

davidruvolo51 commented on data/_ontologies/Resources.csv:
```diff
@@ -1,2 +1,3 @@
-id,name,type
-RD3,Solve-RD RD3,Other type
\ No newline at end of file
+id,name,type,type other,acronym,cohort type,website,start year,end year
```
It's `Resources` in the portal. I'm not sure if that was intentional or leftover from the data model merge.

davidruvolo51 commented:
@mswertz I've updated the demo dataset. The Resources table is now populated with the EGA study metadata and Datasets now has the correct information about the synthetic dataset. I'm tempted to add more information (e.g., collection events, etc.), but we can do that in the next PR when write a script to create the datasets.

mswertz commented on data/_demodata/applications/portal/Biosamples.csv:
```diff
@@ -0,0 +1,19 @@
+id,collected from individual,included in datasets.resource,included in datasets.name
+SAMEA10450711,Case1F,EGAS00001005702,EGAD00001008392
```
is okay. For future we should probably reconsider a long standing discussion on weather we want to treat 'datasets' just like 'resources' having their own identifier (maybe even merge them into one table).

mswertz commented on data/_ontologies/Resources.csv:
```diff
@@ -1,2 +1,3 @@
\ No newline at end of file
+id,name,type,type other,acronym,cohort type,website,start year,end year
+ROR,Research Organization Registry,"Data source,Other type",Ontology,ROR,,https://ror.org/,,
+RD3,Solve-RD,"Data source,Rare disease,Registry",,,Clinical cohort,https://solve-rd.eu,2018,2022
```
this file should not exist in my mind. We don't have an ontology called Resources and we should not have it to not create confusion with the Resources table in our data model.

mswertz reviewed (changes requested):
much better with the study now in play.

However I believe
- _ontologies/Organisations should be proper ontology and maybe simply call it ROR?
- _ontologies/Resources should be deleted

davidruvolo51 commented:
@mswertz I've added the ERNs to the Resources table and deleted the Oragnisations.csv file. I've kept them in the ontologies folder for now. In a future PR, it might be worth restructuring the folders a little bit and adding a `_assets` folder for datasets that aren't ontologies or demo data, but necessary to have in a data model.

davidruvolo51 commented on data/_ontologies/Organisations.csv:
I had to combine the ref and ontology table structures in this dataset so that the loaders wouldn't complain about missing attributes. I really don't like this table structure. In the portal, we typically use organistations as an ontology. It isn't necessary to attach an organisation to a resource. If we have additional data that describes an organisation, then we will extend the ontology table or create a new refback table.

davidruvolo51 replied:
I originally placed the file there, but the FDH models also used the Organisations.csv file as an ontology table. I had two files for organisations (as a ref and as an ontology), but both versions threw a data model loader error. The only way to satisfy the tests was to combine the table structures into a single file.
